planning/obstacle_avoidance_planner/src/utils/utils.cpp

```diff
@@ -708,6 +711,35 @@ bool isOutsideDrivableAreaFromRectangleFootprint(
     tier4_autoware_utils::calcOffsetPose(traj_point.pose, -base_to_rear, -base_to_left, 0.0)
       .position;
 
+  LinearRing2d footprint_polygon;
+  LineString2d left_bound_line;
+  LineString2d right_bound_line;
+  LineString2d back_bound_line;
+
+  footprint_polygon.push_back({top_left_pos.x, top_left_pos.y});
+  footprint_polygon.push_back({top_right_pos.x, top_right_pos.y});
+  footprint_polygon.push_back({bottom_right_pos.x, bottom_right_pos.y});
+  footprint_polygon.push_back({bottom_left_pos.x, bottom_left_pos.y});
+
+  bg::correct(footprint_polygon);
+
+  for (const auto & p : left_bound) {
+    left_bound_line.push_back({p.x, p.y});
+  }
+
+  for (const auto & p : right_bound) {
+    right_bound_line.push_back({p.x, p.y});
+  }
+
+  back_bound_line = {left_bound_line.back(), right_bound_line.back()};
+
+  if (
+    bg::intersects(footprint_polygon, left_bound_line) ||
+    bg::intersects(footprint_polygon, right_bound_line) ||
+    bg::intersects(footprint_polygon, back_bound_line)) {
```

Comment:
You can use bg::intersection(poly1, poly2, output_poly_vec);

Then, this limitation can be resolved.
![image](https://user-images.githubusercontent.com/20228327/213218759-0f257f40-da81-4a8e-a01c-3be8a1c2d459.png)

Reply:
@takayuki5168 Okay, thank you. I thought `intersection` and `intersect` work same way.